### DISCUSSION TITLE
anaconda powershell promt configuration

### DISCUSSION THREAD

**Discussion by AugustHdez** [JanDeDobbeleer/oh-my-posh#4043] [Troubleshoot]
Hello Community,

I've come across a couple of issues while setting up oh-my-posh on anaconda powershell. 

1. anaconda uses PowerShell 5.1 via conda-hook.psm1 and Conda.ps1 found in $USERNAME/anaconda3/shell/condabin. oh-my-posh init pwsh | Invoke-Expression uses PowerShell 7 
2. when I create a new profile.ps1, PowerShell 5.1 recognizes it and loads it leading to an error. 
3. powershell 7, which is my main terminal prompt, becomes slow because it loads the profile I created for Anaconda
4. I tried adding  oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/jandedobbeleer.omp.json" | Invoke-Expression to conda-hook.psm1. While it worked, it still slows PowerShell 7. 

What is the best-suited $PROFILE for Anaconda? and how do I prevent PowerShell 7 and Powershell 5.1 from loading it?

**Comment by JanDeDobbeleer** [comment]
@AugustHdez I get he feeling this is an anaconda question and not oh-my-posh? Or is the oh-my-posh initialisation slowing things down?

**Reply by AugustHdez** [reply]
@JanDeDobbeleer Thank You for the quick response. Yes, it is the oh-my-posh initialization, when an extra profile is added, that slows things down. In this case, when I add a profile for powershel 5.1 or anaconda powershell.

**Reply by JanDeDobbeleer** [reply]
@AugustHdez so, when the initialisation of PowerShell is slower, that's a PowerShell issue (been there for a long time). If the prompt itself becomes slow (press on enter), that's caused by a virus scanner or defender blocking the execution of oh-my-posh and you need to add an exclusion. We have a code signed binary, but even then it still occurs.

**Reply by AugustHdez** [reply]
Thank You for the clarification @JanDeDobbeleer.
